javascript-review.js

```diff
@@ -4,111 +4,209 @@
 
 // Create an object that has four properties. One property should be set so that the value is a number, another property set to a string, the third to an array, and the fourth to a function. This function should simply console.log the value of the first property.
 
+const fourPropObj = {
+  firstKey: 10,
+  secondKey: 'This is a string',
+  thirdKey: ['string', 20],
+  fourthKey: function() {
+    return this.firstKey
+  }
+}
+// console.log(fourPropObj.fourthKey())
+
 // 2.
 
 // Write a constructor function (to create objects) that takes one argument. It should use that argument to set the value of the first attribute of the object. There should be a second attribute that stores the value ‘red’. 
 
 // Note that a constructor function is the old school syntax that was used in javascript before the sugar class syntax came along. 
 
+function Dog(name) {
+  return {
+    name
+  }
+}
+
+// console.log(Dog('Spot'))
+
 // 3.
 
 // Can you access variables defined outside of functions within the scope of functions in JS? Show a simple example of this being used? (Define a constant in global scope and use that in a very simple function.)
 
+const globalNumber = 10
+
+const addFive = (num) => num + 5
+
+// console.log(addFive(globalNumber))
+
```

Comment:
this is more what i was looking for here, in your example you just pass in an argument

```js
const numbers = [1, 2, 3]
const myFunc = () => console.log(numbers)
myFunc()
```